src/Microsoft.DotNet.Darc/src/DarcLib/VirtualMonoRepo/VmrUpdater.cs

```diff
@@ -677,6 +684,87 @@ private string GetCurrentVersion(SourceMapping mapping)
         return version.Sha;
     }
 
+    private async Task CleanUpRemovedRepos(string? readmeTemplatePath, string? tpnTemplatePath)
+    {
+        var deletedRepos = _sourceManifest
+            .Repositories
+            .Where(r => _dependencyTracker.Mappings.FirstOrDefault(m => m.Name == r.Path) == null)
+            .ToList();
+
+        if (!deletedRepos.Any())
+        {
+            return;
+        }
+
+        using var vmrRepository = new Repository(_vmrInfo.VmrPath);
+
+        foreach (var repo in deletedRepos)
+        {
+            _logger.LogWarning("The mapping for {name} was deleted. Removing the repository from the VMR.", repo.Path);
+            var modifiedFiles = DeleteRepository(repo.Path);
+            
+            if(modifiedFiles.Any())
+            {
+                Commands.Stage(vmrRepository, modifiedFiles);
+                Commit($"Delete {repo.Path} from the VMR", DotnetBotCommitSignature);
+            }
+        }
+
+        var sourceManifestPath = _vmrInfo.GetSourceManifestPath();
+        _fileSystem.WriteToFile(sourceManifestPath, _sourceManifest.ToJson());
+
+        if (readmeTemplatePath != null)
+        {
+            await _readmeComponentListGenerator.UpdateReadme(readmeTemplatePath);
+        }
+
+        if (tpnTemplatePath != null)
+        {
+            await _thirdPartyNoticesGenerator.UpdateThirdPartyNotices(tpnTemplatePath);
+        }
+
+        Commands.Stage(vmrRepository,
+            new string[] {
+                sourceManifestPath,
+                _vmrInfo.VmrPath / VmrInfo.ReadmeFileName,
+                _vmrInfo.VmrPath / VmrInfo.ThirdPartyNoticesFileName
+            });
+
+        Commit("Clean up files", DotnetBotCommitSignature);
+    }
+
+    private List<string> DeleteRepository(string repo)
+    {
+        var modifiedFiles = new List<string>();
+
+        var repoSourcesDir = _vmrInfo.GetRepoSourcesPath(repo);
+        try
+        {
+            _fileSystem.DeleteDirectory(repoSourcesDir, true);
+            modifiedFiles.Add(repoSourcesDir);
+            _logger.LogInformation("Deleted directory {dir}", repoSourcesDir);
+        }
+        catch (DirectoryNotFoundException)
+        {
+            _logger.LogInformation("Directory {dir} is already deleted", repoSourcesDir);
+        }
+
+        _sourceManifest.RemoveRepository(repo);
+        _logger.LogInformation("Removed record for repository {name} from {file}", repo, _vmrInfo.GetSourceManifestPath());
+
+        if (_dependencyTracker.RemoveRepositoryVersion(repo))
```

Comment:
How about allversions? We never add that to the modified file list?

Reply:
the whole git-info folder is added